### PR TITLE
Masking password fields for alarm callbacks and outputs.

### PR DESCRIPTION
**Pull request by dennisoelkers**
Fixes Graylog2/graylog2-web-interface#1458

### CODE REVIEW

File: javascript/src/components/outputs/Output.jsx [hunk 1]
```diff
@@ -29,51 +30,69 @@ var Output = React.createClass({
...
     },
+    getInitialState() {
+        return {};
+    },
+    componentDidMount() {
+        this.props.getTypeDefinition(this.props.output.type, (typeDefinition) => {
+            this.setState({typeDefinition: typeDefinition});
+        });
+    },
     /* jshint +W116 */
```
Comment (edmundoa) [inline javascript/src/components/outputs/Output.jsx]: Just a small detail, but this should not be moving around :)

**Comment by edmundoa** [comment]
I saw a problem in this conditional: https://github.com/Graylog2/graylog2-web-interface/pull/1461/files#diff-970bc0e980f6429ea48aa1df32e74607R102

There might be cases where `this.state.outputs` is set but `this.state.assignableOutputs` is still `undefined`, making `AssignOutputDropdown` to raise an exception.

**Comment by edmundoa** [comment]
Other than that looks good to me. I couldn't find any output with password, but as it shares the code with the callback configuration I assume it works.

File: javascript/src/components/outputs/Output.jsx (hunk 1 above)
Comment (dennisoelkers) [inline javascript/src/components/outputs/Output.jsx]: What do you mean?

Comment (edmundoa) [inline javascript/src/components/outputs/Output.jsx]: The jshint comments should be for the `_typeNotAvailable` method. The first one disables the `==` check, and the second one enables it again for the rest of the file. Probably we should have placed the comments inside the method, but anyway :)

Comment (dennisoelkers) [inline javascript/src/components/outputs/Output.jsx]: Got it, thanks!

**Comment by edmundoa** [comment]
Looks good to me :shipit:

**Pull request merged**